presto-spi/src/main/java/com/facebook/presto/spi/function/SqlInvokedFunction.java

```diff
@@ -161,6 +161,15 @@ public long getRequiredVersion()
         return version.get();
     }
 
+    public boolean hasSameDefinitionAs(SqlInvokedFunction that)
```

Comment:
null check for `that` needed here?

Reply:
@ajaygeorge We don't null check method arguments unless we're storing that argument in a field (such as constructors & setters). This is method is also different from `equals` - I think we should throw instead of returning null if `that == null`

Reply:
I'm more thinking of a `checkArgument` rather than return null. It should throw an `IllegalArgumentException` rather than a `NullPointerException`

Reply:
> As a regular instance method, we should throw instead of returning null if `that == null`

as in throw an `IllegalArgumentException` ?

returning `false` for `that==null` seems like a sane thing to do unless we are guaranteed that `that` is never null. 

If there is a pattern we are already using to address these scenarios please use that instead of an explicit null check.

Reply:
As @caithagoras said, we in general do not return null. Very few methods in Presto code base would return null. They either return `Optional` or return an object. It's a good habit so you don't need to check for null in every single place you dereference. Specific to this method, since it's a public taking an arbitrary pointer I think `checkArgument` is still necessary.

Reply:
I wonder what is the special thing in this method that make us want to add a `checkArgument` to the parameter when we almost never do those checks in any other public methods including the ones in spi?

Reply:
Per offline discussion, will add a null check since this method acts similar to null, but avoid allowing null parameter to prevent misuse of this method.